pinot-core/src/main/java/org/apache/pinot/core/data/manager/BaseTableDataManager.java

```diff
@@ -223,8 +224,21 @@ protected void releaseAndRemoveAllSegments() {
       segmentDataManagers = new ArrayList<>(_segmentDataManagerMap.values());
       _segmentDataManagerMap.clear();
     }
-    for (SegmentDataManager segmentDataManager : segmentDataManagers) {
-      releaseSegment(segmentDataManager);
+    if (!segmentDataManagers.isEmpty()) {
+      ExecutorService stopExecutorService = Executors.newFixedThreadPool(segmentDataManagers.size());
```

Comment:
what's the upper bound of total `segmentDataManagers` and is it save to spawn so many threads? Just wondering a smaller limit should be used here

Reply:
+1. We shouldn't create unbounded thread pool. The default should be based on the CPU cores, and we can consider making it configurable